gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java

```diff
@@ -1006,15 +1007,21 @@ public boolean keyUp (InputEvent event, int keycode) {
 			return true;
 		}
 
+		protected boolean checkFocusTraverse(char character) {
+			return focusTraversal && (character == TAB || (
+							(character == CARRIAGE_RETURN || character == NEWLINE)
+							&& (UIUtils.isAndroid || UIUtils.isIos) ));
+		}
+
```

Comment:
I'm not a big fan of this method, especially since it uses the field TextField.focusTraversal, and allows an overriding subclass to ignore the value of this field?

Reply:
@petoncle that is why i implemented it, see TextArea -> other focus traversal behaving.

Reply:
Just sharing an idea, not sure if it's any good: could we maybe have a field `TextField.focusTraversalCharacters`, and populate it in the constructor of TextField like:
```java
focusTraversalCharacters.add(TAB);
if (UIUtils.isAndroid || UIUtils.isIos) {
	focusTraversalCharacters.add(CARRIAGE_RETURN);
	focusTraversalCharacters.add(NEWLINE);
}
```

Reply:
@petoncle, yes that an idea, but the thing is: normally focus traversal depends only those few chars, i think it doesn't make much sense to overextend this.

This would lead to using smileys or something else for focus traverse by adding chars to that list. But i think nobody will implement that :D

Reply:
Maybe `isFocusTraversalCharacter()`?

Reply:
@petoncle, yes that's good, but then i have to remove the focusTraversal-Flag from the method.

Reply:
How about `shouldTraverseFocusOn(char)`?

Reply:
Ok I think I just let it just checkFocusTraverse(char) or checkFocusTraversal(char) that's the reason for my original question. Currently it's called checkFocusTraverse(char).

But thx for the suggestions ;)